core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java

```diff
@@ -56,14 +56,26 @@
   @Override
   public ClientResponse<InputStream> handleResponse(HttpResponse response, TrafficCop trafficCop)
   {
+    ChannelBufferInputStream channelStream = null;
     try {
-      queue.put(new ChannelBufferInputStream(response.getContent()));
+      channelStream = new ChannelBufferInputStream(response.getContent());
+      queue.put(channelStream);
     }
     catch (InterruptedException e) {
       log.error(e, "Queue appending interrupted");
       Thread.currentThread().interrupt();
       throw new RuntimeException(e);
     }
+    finally {
+      if (channelStream != null) {
+        try {
```

Comment:
This block is essentially `CloseQuietly.close()`.

Reply:
Thanks, it is better.